chevah/brink
# [#3435] Add codecov coverage reports.

Scope
=====

This looks at integrating codecov to brink project, via buildbot.

This is the playground for testing codecov and if all is ok we will push it to the other repos.

I want to see how to push multiple coverage reports, for the same commit/PR from multiple slaves

Changes
=======

Add common code to generated and publish coverage.

The coverage is generated only if the CODECOV_TOKEN env var is  set.

This also required updates in our buildmaster to trigger the right things.

The `paver needs_review` command now requires the pull id so that it will know what PR to update.

I will look in another ticket at how to automate this.


How to try and test the changes
===============================

reviewers: @alibotean @brunogola 

check that it looks good.

```
paver test_review 82
```

The whole think is called by buildbot, but you can simulate it on your computer using

To run without coverage make sure you don't have CODECOV_TOKEN defined then run

```
paver test_ci
```

To run with coverage you should get the token from https://codecov.io/gh/chevah/brink/settings

then use

```
export GITHUB_PULL_ID=82
export CODECOV_TOKEN=some_token
paver test_ci
```

Once we have a real PR I will look at the `codecov/changes ` test... this can follow up with a test for improving coverage, at least by updating coveragerc to ignore code which is not going to be executed
